build.rs

```diff
@@ -1,3 +1,5 @@
 fn main() -> shadow_rs::SdResult<()> {
-    shadow_rs::new()
+    shadow_rs::new()?;
+    embed_resource::compile("starship-res.rc");
```

Comment:
The issue in the `embed-resource` crate regarding spurious build failures doesn't sound very good.

Something like this snippet should suffice without `embed-resource`:
```suggestion
    #[cfg(target_env = "msvc")]
    {
        let linker_flags = ["/MANIFEST:EMBED", "/MANIFESTINPUT:starship.exe.manifest"];
        for flag in linker_flags {
            println!("cargo:rustc-link-arg={}", flag);
        }
    }
```

Reply:
Only if you are not going to ever build with GCC or cross compile for Windows. Which might be fine here. We can also use `/MANIFESTUAC` instead of specifying it in the input manifest in this case.

Reply:
I think some people were using the GNU toolchain, so it would be better to support `windows-gnu`. Supporting it will require scripting `windres` which shouldn't be too hard, most of the complexity in other tools seems to be in locating MSVC `rc.exe`.

Instead of custom code, [windres](https://crates.io/crates/windres) also looks like a good alternative to embed-resource, but it's much less popular. [embed-manifest](https://crates.io/crates/embed-manifest) takes an approach that matches my snippet for MSVC, but the code for `windows-gnu` is scary.

Reply:
Tried to see if I can just find the issue in `embed-resource`, I might have found it, or I might not have: https://github.com/nabijaczleweli/rust-embed-resource/issues/20#issuecomment-1040254856

To truly be sure I would need to actually reproduce the error...

Reply:
I think I would still prefer the `windres` crate. It would be nice if rust shipped `llvm-rc` to avoid all this.

Reply:
Also,  it would be nice to `cfg(windows)` the build dependency.